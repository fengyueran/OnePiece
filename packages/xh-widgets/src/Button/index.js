import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import ButtonBase from '../ButtonBase';
import { fade } from '../utils/colorManipulator';

const ButtonStyle = css`
  min-width: 64px;
  min-height: 36px;
  padding: 8px 16px;
  border-radius: 2px;
  margin: 8px;
  font-size: 0.875rem;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-weight: 500;
  line-height: 1.5;
  border-radius: 4px;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  background-color: #e0e0e0; 
  color: rgba(0, 0, 0, 0.87);
  box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 3px 1px -2px rgba(0, 0, 0, 0.12);
  &:hover {
    background-color: ${fade('rgba(0, 0, 0, 0.87)', 0.2)};
  }
`;

const Container = styled.span`
  width: 100%;
  display: inherit;
  align-items: inherit;
  justify-content: inherit;
`;

const Button = ({ 
  isRipple, children, style, className, onClick
}) => (
  <ButtonBase
    onClick={onClick} 
    style={style}
    cssStyle={ButtonStyle}
    className={className}
    isRipple={isRipple}
  >
    <Container>{children}</Container>
  </ButtonBase>
);

Button.propTypes = {
  isRipple: PropTypes.bool,
  style: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.node,
  onClick: PropTypes.func,
};

export { Button };
export default Button;