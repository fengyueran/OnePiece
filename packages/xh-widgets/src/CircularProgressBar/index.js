import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  display: 'inline-block';
  line-height: 1;
`;

const SIZE = 44;

const CircularProgressBar = props => {
  const { size, thickness, value, trailColor, pathColor } = props;
  const rootStyle = {};
  const circleStyle = {};

  rootStyle.transform = 'rotate(-90deg)';
  const circumference = 2 * Math.PI * ((SIZE - thickness) / 2);
  circleStyle.strokeDasharray = circumference.toFixed(3);
  circleStyle.strokeDashoffset = `${(
    ((100 - value) / 100) *
    circumference
  ).toFixed(3)}px`;

  return (
    <Container
      style={{ width: size, height: size, ...rootStyle }}
      role="progressbar"
    >
      <svg viewBox={`${SIZE / 2} ${SIZE / 2} ${SIZE} ${SIZE}`}>
        <circle
          cx={SIZE}
          cy={SIZE}
          r={(SIZE - thickness) / 2}
          fill="none"
          stroke={trailColor}
          strokeWidth={thickness}
        />
        <circle
          style={circleStyle}
          cx={SIZE}
          cy={SIZE}
          r={(SIZE - thickness) / 2}
          fill="none"
          stroke={pathColor}
          strokeWidth={thickness}
        />
      </svg>
    </Container>
  );
};

CircularProgressBar.propTypes = {
  size: PropTypes.number,
  thickness: PropTypes.number,
  value: PropTypes.number,
  pathColor: PropTypes.string,
  trailColor: PropTypes.string
};

CircularProgressBar.defaultProps = {
  size: 40,
  thickness: 3.6,
  value: 0,
  pathColor: '#51B5F4',
  trailColor: '#F5F5F5'
};

export default CircularProgressBar;
