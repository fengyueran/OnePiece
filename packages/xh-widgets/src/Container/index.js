export { 
  FlexContainer,
  VContainer,
  LineContainer
} from './Container';