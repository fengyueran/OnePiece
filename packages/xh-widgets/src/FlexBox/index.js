import styled from 'styled-components';

const FlexBox = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  width: 100%;
`;

export { FlexBox };
export default FlexBox;
