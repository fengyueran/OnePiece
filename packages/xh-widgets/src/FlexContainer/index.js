import styled from 'styled-components';

const FlexContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  width: 100%;
`;

export { FlexContainer };
export default FlexContainer;