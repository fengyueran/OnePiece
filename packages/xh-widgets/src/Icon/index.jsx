import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledIcon = styled.div`
  mask-image: ${props => `url(${props.src})`};
  background: ${props => props.tintColor};
  -webkit-mask-size: contain;
  -webkit-mask-repeat: no-repeat;
  width: ${props => `${props.width}px`};
  height: ${props => `${props.height}px`};
`;

const Icon = props => (
  <StyledIcon {...props} />
);

Icon.propTypes = {
  src: PropTypes.string.isRequired,
  tintColor: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number,
};

Icon.defaultProps = {
  tintColor: '#9520f7',
  width: 20,
  height: 20,
};

export { Icon };
export default Icon;