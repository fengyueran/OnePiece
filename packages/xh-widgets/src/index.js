export * from './FlexContainer';
export * from './LineContainer';
export * from './VContainer';
export * from './CircularProgressBar';
export * from './Button';
export * from './utils';