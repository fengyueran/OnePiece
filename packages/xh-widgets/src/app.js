
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';
import { Button } from './Button';

const StyledButton = styled(Button)`
  background: #536DFE;
`;
class App extends Component {
  render() {
    return (
      <div style={{ padding: 20 }}>
        <StyledButton onClick={() => alert('clicked')}>
          Submit
        </StyledButton>
      </div>
    );
  }
}

ReactDOM.render(<App />, document.getElementById('root'));

export default App;
