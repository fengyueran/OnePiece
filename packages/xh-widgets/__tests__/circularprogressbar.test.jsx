import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import { CircularProgressBar } from '../src/CircularProgressBar';

const setup = (Component, props) => {
  const wrapper = shallow(
    <Component {...props} />
  );
  return wrapper;
};

describe('CircularProgressBar test', () => {
  const props = {
    size: 100,
    thickness: 3,
    value: 50,
    pathColor: 'red',
    trailColor: 'blue',
  };

  const wrapper = setup(CircularProgressBar, props);

  it('CircularProgressBar should be render correctly', () => {
    const svg = wrapper.childAt(0);
    expect(svg.name()).toBe('svg'); 
    expect(svg.childAt(0).name(), 'circle');
    expect(svg.childAt(1).name(), 'circle');
    expect(svg.children().length).toBe(2); 
    expect(svg.childAt(0).props().cx).toBe(44);
    expect(svg.childAt(0).props().stroke).toBe('blue');
    expect(svg.childAt(1).props().stroke).toBe('red');
  });

  it('CircularProgressBar snapshot', () => {
    const tree = renderer
      .create(<CircularProgressBar {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});