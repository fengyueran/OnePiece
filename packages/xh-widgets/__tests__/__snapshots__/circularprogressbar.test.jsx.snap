// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgressBar test CircularProgressBar snapshot 1`] = `
<div
  className="sc-bdVaJa hEvsdq"
  role="progressbar"
  style={
    Object {
      "height": 100,
      "transform": "rotate(-90deg)",
      "width": 100,
    }
  }
>
  <svg
    viewBox="22 22 44 44"
  >
    <circle
      cx={44}
      cy={44}
      fill="none"
      r={20.5}
      stroke="blue"
      strokeWidth={3}
    />
    <circle
      cx={44}
      cy={44}
      fill="none"
      r={20.5}
      stroke="red"
      strokeWidth={3}
      style={
        Object {
          "strokeDasharray": "128.805",
          "strokeDashoffset": "64.403px",
        }
      }
    />
  </svg>
</div>
`;
